app/models/concerns/reviewable_action_builder.rb

```diff
@@ -3,6 +3,49 @@
 module ReviewableActionBuilder
   extend ActiveSupport::Concern
 
+  # Standard post-actions bundle. Consumers should use the returned
+  # bundle value when adding post-focused actions.
+  #
+  # @param actions [Reviewable::Actions] Actions instance to add the bundle to.
+  # @param guardian [Guardian] Guardian instance to check permissions.
+  #
+  # @return [Reviewable::Actions::Bundle] The created post actions bundle.
+  def build_post_actions_bundle(actions, guardian)
+    bundle =
+      actions.add_bundle(
+        "#{id}-post-actions",
+        label: "reviewables.actions.post_actions.bundle_title",
+      )
+
+    # Always include the no-op action
+    build_action(actions, :no_action_post, bundle:)
+
+    return bundle unless target_post
+
+    if target_post.trashed? && guardian.can_recover_post?(target_post)
+      build_action(actions, :restore_post, bundle:)
+    end
+
+    if target_post.hidden?
+      build_action(actions, :unhide_post, bundle:) if !target_post.user_deleted?
+    else
+      build_action(actions, :hide_post, bundle:)
+    end
+
+    if guardian.can_delete_post_or_topic?(target_post)
+      build_action(actions, :delete_post, bundle:)
+      if target_post.reply_count > 0
```

Comment:
I see an opportunity to add this as a method on `Post`, to help people avoid doing something like `post.replies.any?` and accidentally incurring a performance penalty.

Reply:
I haven't dug into writing rubocop rules before, but it seems like that kind of thing could be detected and warned about in the [Discourse rules](https://github.com/discourse/rubocop-discourse/).